scene/2d/parallax_2d.cpp

```diff
@@ -72,7 +72,13 @@ void Parallax2D::_validate_property(PropertyInfo &p_property) const {
 
 void Parallax2D::_camera_moved(const Transform2D &p_transform, const Point2 &p_screen_offset, const Point2 &p_adj_screen_pos) {
 	if (!ignore_camera_scroll) {
-		set_screen_offset(p_adj_screen_pos);
+		Point2 adj_screen_pos = p_adj_screen_pos;
+
+		if (GLOBAL_GET("rendering/2d/snap/snap_2d_transforms_to_pixel")) {
```

Comment:
This should probably be cached somehow, it is expensive to check every time

Reply:
Should be `get_viewport() && get_viewport()->is_snap_2d_transforms_to_pixel_enabled()` to get the actual setting of the current viewport.

Reply:
Yeah, I don't know of any notification for settings changed, so alvin's is the only optimization I can think of. Which is a shame because we also use `GLOBAL_GET` for the viewport size... but that's more forgivable since it's only in the editor.

Reply:
There's the signal in settings, used in a number of places like:
https://github.com/godotengine/godot/blob/f3af22b10b1e64146d48be7726f6395c9a4185e9/scene/main/window.cpp#L1312-L1313

Reply:
Ooh. Okay, maybe that'd be better to do, then. I'll make a separate PR for the viewport size.

Reply:
I'm going to leave it with alvin's suggestion for this one since you can change the 2D snap setting on the fly with `SubViewport` without going into the project settings. The other calls I can cache because it's an editor only feature and we don't expose a common way to get a viewport's size right now afaik.